app/grandchallenge/algorithms/serializers.py

```diff
@@ -90,3 +94,85 @@ class HyperlinkedJobSerializer(JobSerializer):
 
     class Meta(JobSerializer.Meta):
         pass
+
+
+class JobPostSerializer(JobSerializer):
+    algorithm_title = CharField(write_only=True)
```

Comment:
You shouldn't select algorithms by title as these are not unique, use slug instead, and I think that there is a SlugField which also does the validation. We also need to ensure that the permissions are checked correctly.